File: models/dance.js
import mongoose from 'mongoose';


const danceSchema = new mongoose.Schema({
  id: {
    type: Number,
    required: true,
    unique: true,
    index: true
  },
  name: {
    type: String,
    required: true,
    trim: true
  },
  type: {
    type: String,
    trim: true,
    
  },
  origin: {
    type: String,
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  imageurl: {
    type: String,
    default: ""
    
  },
  videourl: {
    type: String,
    default: ""
    
  },
  wikiurl: {
    type: String,
    trim: true,
    
  }
}, { timestamps: true });




const Dance = mongoose.model('Dance', danceSchema);
export default Dance;

File: frontend/app/src/main/java/com/example/richculture/Data/Dance.kt
package com.example.richculture.Data


data class Dance(
    val _id: String,
    val id: Int,
    val name: String,
    val type: String,
    val origin: String,
    val description: String,
    val imageurl: String,
    val videourl: String?,   // Sometimes empty
    val wikiurl: String,
    val __v: Int,
    val createdAt: String,
    val updatedAt: String
)
